avr-hal-generic/src/i2c.rs

```diff
@@ -1,11 +1,22 @@
 //! I2C Implementations
 
+// datasheet : ATmega48A-PA-88A-PA-168A-PA-328-P-DS-DS40002061B.pdf
```

Comment:
I added a bunch of datasheet excerpts as comments as a way to help me understand what's going on. Personally, I like lot's of explanatory comments, but of course, it's up to you to decide if you want to keep them in the final PR.

Reply:
I quite like what the kernel has to say about commenting: https://www.kernel.org/doc/html/latest/process/coding-style.html#commenting

I.e. the HOW should be obvious from the code and shouldn't need a datasheet excerpt alongside it.  For tricky stuff, mentioning the relevant section of the datasheet might make sense but I'd never copy the contents into the code -> if I need them, I can just open the data-sheet.

Reply:
Funny, I was coming around to this opinion myself in the past few days. Adding in the comments helped me think about the problem for awhile, but then they started getting in the way. It _would_ be nice if the datasheets were deep-linkable, so that I could reference the line number of the spec. I'll check in my latest changes in a few minutes.

Reply:
> It would be nice if the datasheets were deep-linkable

Yeah, it really would ... Best one can do is name the section _and_ revision of the datasheet (because some lunatics think it is a good idea to renumber the sections from time to time ...)

> Adding in the comments helped me think about the problem for awhile, but then they started getting in the way.

In the C world some people have adopted a nice method: Use `//` comments for any temporary in-development comments while thinking/developing/debugging.  Use `/* ... */` comments for "production" comments that are meant to stay.  Then, before submitting changes, remove all `//` comments.

In Rust, this doesn't really work because people just use `//` for all comments everywhere.  So I personally have started to just use some comment prefix to mark 'dev-comments', e.g. `// WIP: ...`.

Reply:
+1 on the `// WIP : ...` I typically add either `// TODO ...` or `// BUGBUG ...` but I'll adopt `WIP` for temp in dev comments going forward.